ion-schema/src/result/mod.rs

```diff
@@ -3,62 +3,92 @@
 //! [`IonSchemaResult<T, E>`][`IonSchemaResult`]  is the type used for returning and propagating errors.
 //! It is an enum with the variants, Ok(T), representing success and containing a value,and Err(E), representing an [`IonSchemaError`].
 
+mod invalid_schema_error;
+mod isl_source_location;
+
+pub use invalid_schema_error::InvalidSchemaError;
+pub(crate) use isl_source_location::*;
+
 use crate::violation::Violation;
 use ion_rs::IonError;
+use std::convert::Infallible;
+use std::fmt::{Debug, Write};
```

Comment:
I'll fix this before merging.
```suggestion
use std::fmt::Debug;
```